pkg/plugin/plugin.go

```diff
@@ -78,6 +82,55 @@ type Info struct {
 	ConfigAttributes json.RawMessage `db:"config_attrs" json:"config_attrs"` // ConfigOption(s) as json-encoded list
 }
 
+// NewInfo constructs a new Info to be used in Plugin.GetInfo.
+//
+// The configOptions will be enriched with defaults from the typePtr, being a pointer to the Plugin implementation.
+func NewInfo(typePtr interface{}, name, version, author string, configOptions []*ConfigOption) (*Info, error) {
+	err := defaults.Set(typePtr)
+	if err != nil {
+		return nil, err
+	}
+
+	typeDefaults := make(map[string]any)
+	typeRefT := reflect.TypeOf(typePtr).Elem()
+	typeRefV := reflect.ValueOf(typePtr)
+	for i := 0; i < typeRefT.NumField(); i++ {
+		if !typeRefT.Field(i).IsExported() {
+			continue
+		}
+
+		jsonVal, jsonOk := typeRefT.Field(i).Tag.Lookup("json")
+		_, defaultOk := typeRefT.Field(i).Tag.Lookup("default")
+
+		if !jsonOk {
+			return nil, fmt.Errorf("exported field %q misses json struct tag, set to \"-\" to ignore", typeRefT.Field(i).Name)
+		}
```

Comment:
Is it really necessary to fail with an error here? Why not just skip it?

Reply:
As this function is only being called on known channel plugins, I would like them to fail hard in case something is missing. Otherwise this might only lead to strange errors.